mock/py/mock-hermetic-repo.py

```diff
@@ -23,6 +26,33 @@
 log = logging.getLogger(__name__)
 
 
+def retry(timeout=300, interval=15, jitter=0.2, retry_on=Exception):
+    """A decorator that retries a section of code until success or timeout."""
+
+    def wrapper(function):
+        @functools.wraps(function)
+        def inner(*args, **kwargs):
+            start = time.time()
+            while True:
+                try:
+                    return function(*args, **kwargs)
+                except retry_on as e:
+                    skew = random.random() * jitter + (1 - jitter / 2)
+                    print(
+                        "Exception %r raised from %r.  Retry in %rs"
+                        % (e, function, interval * skew),
+                        file=sys.stderr,
+                    )
+                    time.sleep(interval * skew)
+                    if (time.time() - start) >= timeout:
+                        raise  # This re-raises the last exception.
+
+        return inner
+
+    return wrapper
+
+
+@retry(retry_on=requests.exceptions.ConnectionError)
```

Comment:
Have you considered python-backoff?  Mock already depends on it so it is not a new dep:
https://github.com/rpm-software-management/mock/blob/5193dbd1ea984688b7c11ecfdbf4baa8ed8520fc/mock/py/mockbuild/file_downloader.py#L67-L68